internal/app/go-filecoin/plumbing/api.go

```diff
@@ -250,11 +250,14 @@ func (api *API) MessagePreview(ctx context.Context, from, to address.Address, me
 // it does not change any state. It is use to interrogate actor state. The from address
 // is optional; if not provided, an address will be chosen from the node's wallet.
 func (api *API) MessageQuery(ctx context.Context, optFrom, to address.Address, method types.MethodID, baseKey block.TipSetKey, params ...interface{}) ([][]byte, error) {
-	snapshot, err := api.actorState.Snapshot(ctx, baseKey)
-	if err != nil {
-		return [][]byte{}, err
-	}
-	return snapshot.Query(ctx, optFrom, to, method, params...)
+	// Dragons: delete
```

Comment:
Are you suggesting deleting the concept of message querying from plumbing (sounds like more trouble than its worth) or suggesting deleting the snapshot based implementation?

Reply:
This got dropped a while a go. The actors do not have support for it. 
The way to access state is directly through each actors state object.